db/view/view.cc

```diff
@@ -2626,6 +2626,27 @@ update_backlog node_update_backlog::add_fetch(unsigned shard, update_backlog bac
     return std::max(backlog, _max.load(std::memory_order_relaxed));
 }
 
+update_backlog node_update_backlog::add_fetch_for_response(unsigned shard, update_backlog backlog) {
+    _need_publishing[shard].store(need_publishing::yes, std::memory_order_relaxed);
+    return add_fetch(shard, backlog);
+}
+
+std::optional<update_backlog> node_update_backlog::add_fetch_for_gossip(unsigned shard, update_backlog backlog, std::optional<db::view::update_backlog> last_backlog) {
+    auto need_publishing = need_publishing::no;
+    for (auto& np : _need_publishing) {
+        if (need_publishing) {
+            break;
+        }
+        need_publishing = np.exchange(need_publishing::no, std::memory_order_relaxed);
+    }
+    auto max = add_fetch(shard, backlog);
+    if (!need_publishing && last_backlog && *last_backlog == max) {
+        return std::nullopt;
+    } else {
+        return max;
+    }
+}
```

Comment:
Do we have to check agains the last backlog at all? If the value has changed then one of the `need_publishing` flags must have been updated, no?